src/main.rs

```diff
@@ -1,21 +1,63 @@
+//! Default Compute@Edge template program.
+
+use fastly::http::response::Builder as ResponseBuilder;
 use fastly::http::{HeaderValue, Method, StatusCode};
 use fastly::request::CacheOverride;
 use fastly::{downstream_request, Body, Error, Request, RequestExt, Response, ResponseExt};
 use std::convert::TryFrom;
 
-const VALID_METHODS: [Method; 3] = [Method::HEAD, Method::GET, Method::POST];
+/// The name of a backend server associated with this service.
+///
+/// This should be changed to match the name of your own backend. See the the `Hosts` section of
+/// the Fastly WASM service UI for more information.
+const BACKEND_NAME: &str = "backend_name";
+
+/// The name of a second backend associated with this service.
+const OTHER_BACKEND_NAME: &str = "other_backend_name";
+
+/// The entrypoint for your application.
+///
+/// This function is triggered when your service receives a client request. This function can call
+/// [`fastly::downstream_request`][dsreq] to get the client request, and should ultimately call
+/// [`fastly::send_downstream`][send] on a [`fastly::Response`][resp] to deliver an HTTP response
+/// to the client.
+///
+/// If `main` returns an error, a [500 Internal Server Error][err] will be delivered to the client.
+///
+/// [dsreq]: ../fastly/request/fn.downstream_request.html
+/// [err]: https://tools.ietf.org/html/rfc7231#section-6.6.1
+/// [resp]: ../fastly/struct.Response.html
+/// [send]: ../fastly/response/trait.ResponseExt.html#method.send_downstream
```

Comment:
Not super keen on this markdowny syntax - does this output nice docs somewhere?  My feeling is that the customer is likely to be interacting with this content *directly*, rather than viewing a rendered version of it somewhere else - so it may be better to refer to syntax more literally (otherwise I might be tempted to write something like

```
let req = [`fastly::downstream_request`]();
```

Reply:
Rustdoc, rust's documentation generating backend, is a explicitly a markdown renderer. This style of linking docs is pretty common in the rust ecosystem.  

https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#basic-structure 

Rustdoc is built into the rust toolchain. A common workflow for browsing documentation for a repo is running `cargo doc --open` to generate and open docs in your local browser.

Rust customers will be familiar with this. Customers new to rust might benefit from a readme note on rust workflows.

Reply:
Thanks for pointing this out @triblondon, these are really good points! Folks are almost always going to be interacting with these comments directly as source. While addressing this, I realized something else: we don't export `main`, so the output in `cargo doc` won't include our functions anyhow. Ex:

![default-program-docs](https://user-images.githubusercontent.com/57912822/79788871-50204500-8317-11ea-8657-bea92d239fb1.png)

So, fancy comments really don't do much good here. Removed!